htlcswitch/hop/payload.go

```diff
@@ -136,8 +136,11 @@ func NewLegacyPayload(f *sphinx.HopData) *Payload {
 // NewPayloadFromReader builds a new Hop from the passed io.Reader. The reader
 // should correspond to the bytes encapsulated in a TLV onion payload. The
 // final hop bool signals that this payload was the final packet parsed by
-// sphinx.
-func NewPayloadFromReader(r io.Reader, finalHop bool) (*Payload, error) {
+// sphinx. A blinding kit is passed in to help handle payloads that are part
+// of a blinded route.
+func NewPayloadFromReader(r io.Reader, finalHop bool,
+	blindingKit *BlindingKit) (*Payload, error) {
```

Comment:
Alternatively, it can just parse everything as normal, then if the payload has a blinded path, the caller handles the processing of it. 

Otherwise, we have a function that right now _just_ does simple byte decoding, but then will _also_ need to do crypto operations.

Reply:
> we have a function that right now just does simple byte decoding

It also does TLV payload "field checking" where we validate that we've got the right TLV types, which made me want to put checks for the encrypted data blob here as well.

Gone with your suggestion of processing the `encrypted_data` separately, only downside is that we need to surface the `tlvMap` with parsed types so that we can check that we have valid TLVs _accross_ the payload and the blob but I think it still turns out cleaner!